models/maespa/R/met2model.MAESPA.R

```diff
@@ -0,0 +1,265 @@
+  end_year <- year(end.date)
+  
+  ## loop over files
+  for(year in start_year:end_year){
+    print(year)
+    
+    old.file <- file.path(in.path, paste(in.prefix, year, "nc", sep="."))
+    
+    if(file.exists(old.file)){
+      ## open netcdf
+      nc <- nc_open(old.file)
+      ## convert time to seconds
+      sec   <- nc$dim$time$vals  
+      sec = udunits2::ud.convert(sec,unlist(strsplit(nc$dim$time$units," "))[1],"seconds")
+      
+      ifelse(leap_year(year)==TRUE,
+             dt <- (366*24*60*60)/length(sec), #leap year
+             dt <- (365*24*60*60)/length(sec)) #non-leap year
+      tstep = round(86400/dt)
+      dt = 86400/tstep
+      
+      #Check wich variabales are available and which are not
+      
+      ## extract variables
+      lat  <- ncvar_get(nc,"latitude")
+      lon  <- ncvar_get(nc,"longitude")       
+      #RAD <-  ncvar_get(nc,"surface_downwelling_shortwave_flux_in_air") #W m-2
+      PAR <-   try(ncvar_get(nc,"surface_downwelling_photosynthetic_photon_flux_in_air")) #mol m-2 s-1
+      TAIR <-  ncvar_get(nc,"air_temperature") #K
+      #TSOIL <-  try(ncvar_get(nc,"soil_temperature")) #K
+      #RHperc <-  try(ncvar_get(nc,"relative_humidity_percentage")) 
+      `RH%` <- try(ncvar_get(nc,"relative_humidity"))
+      #VMFD <-  try(ncvar_get(nc,"vapour_pressure_mole_fraction_deficit")) 
+      #SH <- ncvar_get(nc,"specific_humidity") #
+      PPT <-  ncvar_get(nc,"precipitation_flux") #kg m-2 s-1
+      #WIND <- try(ncvar_get(nc,"wind_speed")) #m/s
+      #PRESS <-  ncvar_get(nc,"air_pressure")# Pa
+      #SW <-  try(ncvar_get(nc,"soil_moisture"))# kg m-2
+      CA <- try(ncvar_get(nc,"mole_fraction_of_carbon_dioxide_in_air")) #mol/mol
+      ##############
+      # Not part of example being used for testing. Needs to be active at some point though:
+      # SWP <- ncvar_get(nc,"soil_water_potential)# MPa
+      # TDEW <- try(ncvar_get(nc, "dew_point_temperature")) #Celsius
+      # FBEAM <- ncvar_get try((nc,"fraction_of_surface_downwelling_photosynthetic_photon_flux_in_air"))#fraction
+      # RH <- try(ncvar_get(nc,"relative_humidity"))# percentage
+      # RAD<- try(ncvar_get(nc,"surface_downwelling_shortwave_flux_in_air))#Wm-2
+      ##############
+      
+      #mmol m-2s-1   hourly incident photosynthetically active radiation. If nonexistant Calculated from RAD.If RAD absent, calculated from TAIR (DELTAT needed).
+      if(!is.numeric(PAR)==TRUE){
+        print("Photosynthetically Active Radiation(PAR)Calculated from RAD.If RAD absent, calculated from TAIR (DELTAT needed).")}else{
+          #4.6= unit conversion of W/m2 to mmol/m2/s
+          #0.4805 = based on the approximation that PAR is 0.45-0.50 of the total radiation
+          PAR <- round(PAR*4.6*0.4805,digits=3) ##Convert W/m2 to mmol/m2/s 
+        }
+      # ºC   air temperature. If nonexistant. Error.
+      TAIR <- udunits2::ud.convert(TAIR,"kelvin","celsius") 
+      TAIR<-round(TAIR,digits =3)
```

Comment:
Why are you rounding TAIR? If there's a restriction on the output format, that needs to be handled in the outputting -- just rounding the data doesn't necessarily change the number of digits written to file.